File: 250324/크기가 N인 순열/n-permutation.js
const fs = require("fs");
const input = fs.readFileSync(0).toString().trim().split('\n');

const n = Number(input[0]);

// Please Write your code here.

const q = []
const visited = Array(n+1).fill(false)
const Permutation = () => {
    if(q.length === n) {
        console.log(q.join(' '))
        return;
    }

    for(let i = 1; i<=n; i++) {
        if(visited[i]) continue;

        q.push(i);
        visited[i] = true;
        Permutation();
        visited[i] = false;
        q.pop();
    }
}

Permutation()

File: 250927/크기가 N인 순열/n-permutation.java
import java.util.*;

public class Main {
    static int n;
    static boolean[] visited;
    static List<Integer> list = new ArrayList<>();

    public static void main(String[] args) {
        // Please write your code here.
        Scanner sc = new Scanner(System.in);

        n = sc.nextInt();
        visited = new boolean[n+1];
        Backtracking();
    }

    private static void Backtracking() {
        if(list.size() == n) {
            list.forEach(v -> System.out.print(v + " "));
            System.out.println();
            return;
        }

        for(int i = 1; i<=n; i++) {
            if(visited[i]) continue;

            visited[i] = true;
            list.add(i);
            Backtracking();
            list.remove(list.size() - 1);
            visited[i] = false;
        }
    }
}